### PR TITLE
added metasploit http DoS module

### PR DESCRIPTION
**Pull request by deepsight**
Adding a module for for CVE-2019-5645, Denial of Service on Metasploit framework HTTP(s) handler.

### CODE REVIEW

**Comment by jmartin-r7** [comment]
Please update the disclosure date to ISO 8601 format as `2019-09-04`.

```
[*] Running msftidy.rb in ./.git/hooks/post-merge mode
--- Checking new and changed module syntax with tools/dev/msftidy.rb ---
modules/auxiliary/dos/http/metasploit_httphandler_dos.rb - [ERROR] Incorrect disclosure date format
modules/auxiliary/dos/http/metasploit_httphandler_dos.rb - [WARNING] Please add a newline at the end of the file
```

File: modules/auxiliary/dos/http/metasploit_httphandler_dos.rb [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                      'License' => MSF_LICENSE,
+                      'References' => [
+                          ['CVE', '2019-5645']
+                      ],
+                      'DisclosureDate' => '2019-09-04'
+          ))
+
+    register_options(
+        [
+            OptString.new('DOSTYPE', [true, 'GENTLE|SOFT|HARD', 'HARD'])
```
Comment (bcoles) [inline modules/auxiliary/dos/http/metasploit_httphandler_dos.rb]:
```suggestion
            OptEnum.new('DOSTYPE', [true, 'HARD', %w[GENTLE SOFT HARD]])
```

File: modules/auxiliary/dos/http/metasploit_httphandler_dos.rb [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      )
+
+      if resthree.body.length == 0
+        print_good('SUCCESS, Service not responding.')
+      else
+        print_error('Service responded with a valid HTTP Response; Attack failed.')
+      end
+
+    else
+      bla = ""
```
Comment (bcoles) [inline modules/auxiliary/dos/http/metasploit_httphandler_dos.rb]:
```suggestion
      fail_with Failure::BadConfig, 'Invalid DOSTYPE selected'
```

File: modules/auxiliary/dos/http/metasploit_httphandler_dos.rb [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    ensure
+      disconnect
+    end
+    true
+  end
+
+  def run
+    print_status("#{rhost}:#{rport} - Sending DoS packet...")
+    dos
+
```
Comment (bcoles) [inline modules/auxiliary/dos/http/metasploit_httphandler_dos.rb]:
```suggestion
```

File: modules/auxiliary/dos/http/metasploit_httphandler_dos.rb [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Dos
+
+  def initialize(info = {})
+    super(update_info(info,
+                      'Name' => 'Metasploit HTTP(S) handler DoS',
```
Comment (bcoles) [inline modules/auxiliary/dos/http/metasploit_httphandler_dos.rb]: Two spaces for indentation please, despite what Rubocop says.

**Comment by busterb** [comment]
Updated options to take an enum, fixed some other minor things, verified function, and added module docs. Thanks @deepsight and @bcoles for the review.

**Comment by busterb** [comment]
## Release Notes

This adds a DoS module targeting a weakness in regex parsing in Metasploit 5.0.27 and earlier's reverse_http and reverse_https payload handlers.

**Comment by jirayutza1** [comment]
> Please update the disclosure date to ISO 8601 format as `2019-09-04`.
> 
> ```
> [*] Running msftidy.rb in ./.git/hooks/post-merge mode
> --- Checking new and changed module syntax with tools/dev/msftidy.rb ---
> modules/auxiliary/dos/http/metasploit_httphandler_dos.rb - [ERROR] Incorrect disclosure date format
> modules/auxiliary/dos/http/metasploit_httphandler_dos.rb - [WARNING] Please add a newline at the end of the file
> ```



> Please update the disclosure date to ISO 8601 format as `2019-09-04`.
> 
> ```
> [*] Running msftidy.rb in ./.git/hooks/post-merge mode
> --- Checking new and changed module syntax with tools/dev/msftidy.rb ---
> modules/auxiliary/dos/http/metasploit_httphandler_dos.rb - [ERROR] Incorrect disclosure date format
> modules/auxiliary/dos/http/metasploit_httphandler_dos.rb - [WARNING] Please add a newline at the end of the file
> ```

**Pull request merged**